sematic/ui/src/components/PipelineBar.tsx

```diff
@@ -86,143 +163,47 @@ export default function PipelineBar(props: {
     []
   );
 
-  const fetchResolution = useCallback(
-    (run: Run | undefined) => {
-      if (run) {
-        fetchJSON({
-          url: "/api/v1/resolutions/" + run.id,
-          apiKey: user?.api_key,
-          callback: (response: ResolutionPayload) => {
-            setResolution(response.content);
-          },
-          setError: (_: Error | undefined) => {
-            // this means the pipeline resolution failed
-            setResolution(undefined);
-          },
-        });
-      }
-    },
-    [setResolution]
-  );
```

Comment:
Moving state management of resolutions and root runs to the `PipelineView` component as the resolution needs to be propagated down the `RunPanel` as well in order to figure out if a run can be re-run (resolution has container image uri).